Jint/Engine.cs

```diff
@@ -162,6 +167,39 @@ public Engine(Action<Options> options)
             ClrTypeConverter = new DefaultTypeConverter(this);
             BreakPoints = new List<BreakPoint>();
             DebugHandler = new DebugHandler(this);
+
+            if (Options._ExtensionMethodsTypes.Any())
```

Comment:
Can you move it to a private method?

Also it's quite some expensive code to run on each engine instance. At some point we might want to cache the dictionary statically (and keeping it thread-safe).